aptos-move/framework/aptos-stdlib/sources/string_utils.move

```diff
@@ -0,0 +1,125 @@
+module aptos_std::string_utils {
+    use std::string::String;
+
+    const EARGS_MISMATCH: u64 = 1;
+    const EINVALID_FORMAT: u64 = 2;
+
+    struct List<T, N> has copy, drop, store {
+        car: T,
+        cdr: N,
+    }
+
+    struct NIL has copy, drop, store {}
+
+    public fun cons<T, N>(car: T, cdr: N): List<T, N> {
+        List { car, cdr }
+    }
+
+    public fun nil(): NIL { NIL {} }
+
+    native fun format<T>(s: &T): String;
```

Comment:
Can you add documentations regarding what these do? Also shouldn't these be public?